framework/assets/yii.validation.js

```diff
@@ -436,5 +436,32 @@ yii.validation = (function ($) {
         }
     }
 
+    function validateMimeType(mimeTypes, fileType)
+    {
+        for (var i=0; i<mimeTypes.length; i++) {
+            if (mimeTypes[i].indexOf('*') !== -1 && validateMimeTypeMask(mimeTypes[i], fileType)) {
+                return true;
+            }
+
+            if (mimeTypes[i] == fileType) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    function validateMimeTypeMask(mimeTypeMask, fileType)
+    {
+        var escapedString = escapeRegExp(mimeTypeMask);
+        var regexp = new RegExp('^' + escapedString.replace('\\*', '.*') +  '$');
+
+        return regexp.test(fileType);
+    }
+
+    function escapeRegExp(str) {
+        return str.replace(/[-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&");
```

Comment:
I guess replace at https://github.com/yiisoft/yii2/pull/10393/files#diff-56c3babe983e0fddaebd160c0bf620bdR457 is needed because of extra slashes...